livekit-agents/livekit/agents/beta/tools/http/send_http.py

```diff
@@ -0,0 +1,306 @@
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+import aiohttp
+
+from .... import FunctionTool
+from ....llm.tool_context import ToolError, function_tool
+from ....log import logger
+from ....voice import AgentSession, RunContext
+
+
+@dataclass
+class HTTPResponse:
+    """Structured response from HTTP request.
+
+    Attributes:
+        success: Whether the request was successful (2xx status)
+        status_code: HTTP status code
+        body: Response body as string
+        headers: Response headers
+        error: Error message (if failed)
+    """
+
+    success: bool
+    status_code: int
+    body: str
+    headers: dict[str, str]
+    error: str | None = None
+
+
+@dataclass
+class HTTPToolConfig:
+    """Configuration for HTTP tool customization.
+
+    Attributes:
+        url: Endpoint URL (required)
+        name: Custom tool name (optional)
+        description: Custom tool description (optional)
+        method: HTTP method (default: "POST")
+        headers: Optional dict of headers to send with request
+        timeout_ms: Request timeout in milliseconds (default: 30000)
+        parameters: JSON schema dict for tool parameters (defaults to empty object schema)
+        execution_message: Optional message to announce before executing (spoken by agent)
+        output_normalizer: Optional function to transform HTTPResponse to string or None
```

Comment:
rather than adding additional complexity to the tool - it may be better to create two layers of abstraction:

1. a simplified utility for performing a URL request from a tool call that handles variable templating and errors consistently, and could be used inside any function_tool
2. a tool factory method that takes an HTTP config and outputs a simple tool using the aforementioned utility

this way devs could add any kind of before/after customization they need by leveraging the lower-level utility, but we'd still offer a no-frills basic wrapper